### PR TITLE
implemented delete route and unit tests

### PR DESCRIPTION
**Pull request by GrunWong**
Part of #41 

## Proposed Changes

  -Add delete route for posts/{id}
  -Add testing for delete route in implementation

### CODE REVIEW

File: server/test/integration/post.spec.js [hunk 1]
```diff
@@ -106,4 +106,81 @@ describe('Posts API', () => {
...
+
+    const createdPost = response.body;
+    const url = '/posts/';
+
+    const response1 = await supertest(app).delete(url.concat(createdPost._id));
+
+    expect(response1.status).toBe(200);
+
+    const response3 = await supertest(app).get(url);
+    console.log(response3.body);
```
Comment (kaplunket) [inline server/test/integration/post.spec.js]: Remove?

File: server/test/integration/post.spec.js [hunk 2]
```diff
@@ -106,4 +106,81 @@ describe('Posts API', () => {
...
+    expect(response.status).toBe(201);
+
+    const createdPost = response.body;
+    const url = '/posts/';
+
+    const response1 = await supertest(app).delete(url.concat(createdPost._id));
+
+    expect(response1.status).toBe(200);
+
+    const response3 = await supertest(app).get(url);
```
Comment (ErnWong) [inline server/test/integration/post.spec.js]: Is this required by the test? The response3 variable is not used anywhere apart from the console log.

File: server/test/integration/post.spec.js [hunk 3]
```diff
@@ -106,4 +106,81 @@ describe('Posts API', () => {
...
+    const createdPost = response.body;
+    const url = '/posts/';
+
+    const response1 = await supertest(app).delete(url.concat(createdPost._id));
+
+    expect(response1.status).toBe(200);
+    done();
+  });
+
+  it('tests the deletion of already deleted post and return error message', async done => {
```
Comment (ErnWong) [inline server/test/integration/post.spec.js]: Where is it testing that it returns an error message?

File: server/test/integration/post.spec.js (hunk 2 above)
Comment (GrunWong) [reply server/test/integration/post.spec.js]: I will change that! Good spotting, it was used for me to see that the deletion actually went through.

File: server/test/integration/post.spec.js (hunk 3 above)
Comment (GrunWong) [reply server/test/integration/post.spec.js]: Ah yes, nice one. That is a typo, it's meant to read "returns as error" instead.

File: server/test/integration/post.spec.js (hunk 1 above)
Comment (GrunWong) [reply server/test/integration/post.spec.js]: Console.log has now been removed.

File: server/test/integration/post.spec.js [hunk 4]
```diff
@@ -106,4 +106,80 @@ describe('Posts API', () => {
...
+    const createdPost = response.body;
+    const url = '/posts/';
+
+    const response1 = await supertest(app).delete(url.concat(createdPost._id));
+
+    expect(response1.status).toBe(200);
+    done();
+  });
+
+  it('tests the deletion of already deleted post and returns as error', async done => {
```
Comment (ErnWong) [inline server/test/integration/post.spec.js]: Where is it testing that it is returning an error?

File: server/test/integration/post.spec.js [hunk 5]
```diff
@@ -106,4 +106,80 @@ describe('Posts API', () => {
...
+      .post('/posts')
+      .send(postData);
+
+    const createdPost = response.body;
+    const url = '/posts/';
+
+    const response1 = await supertest(app).delete(url.concat(createdPost._id));
+
+    expect(response1.status).toBe(200);
+    done();
```
Comment (ErnWong) [inline server/test/integration/post.spec.js]: Should this also test that the post can longer be retrieved after it is deleted?

**Pull request merged**